File: src/Week02/Ts/singleton.ts
class Singleton {
    private constructor(){ }

    private static instance:Singleton=new Singleton();

    public static getInstance():Singleton{
        return Singleton.instance;
    }

    public hotelList:string[]=[];
}

console.log("Bob selected XY hotel");
let bobObj:Singleton=Singleton.getInstance();
bobObj.hotelList.push("XY");
console.log(bobObj.hotelList);  //--> [ 'XY' ]

console.log("Alice selected WZ hotel");
let aliceObj:Singleton=Singleton.getInstance();
aliceObj.hotelList.push("WZ");
console.log(aliceObj.hotelList); //--> [ 'XY', 'WZ' ]

File: src/Week02/Py/Singleton.py
class Singleton:

    _instance = None
    def __init__(self):
        if not Singleton._instance:
            print("Creating new instance")
            Singleton.instance = self
        else:
            raise Exception(Singleton._instance)

def get_instance():
    instance=None
    try:
        instance = Singleton()
    except Exception as e:
        instance = e
    return instance
 

if __name__ == "__main__":
    s1 = get_instance()
    print(s1)
    s2 = get_instance()
    print(s2)
    Singleton._instance = None    
    s3 = get_instance()
    print(s3)